### ISSUE TITLE
Consider new social media integrations

### ISSUE THREAD

**Issue by willguv** [localgovdrupal/localgov_elections#141]
As Twitter/ X is falling out of favour, we need to consider other social media services

Many public sector people have moved to BlueSky, most already have accounts on Linked In. Threads has also been mentioned.

We need to:
- consider the next best service to integrate with
- check their APIs
- estimate a new integration
- raise funding if needed/ built it

**Comment by markconroy** [comment]
I don't think this feature should be part of the LocalGov Elections module.

I think it should be a generic module that sites can add to any/all content types. For example, I can imagine this being useful for news and events.

To achieve this, I'd suggest we use the [Drupal Social API](https://www.drupal.org/project/social_api) module which is actively maintained, is installed on nearly 6,000 websites, and had a new release as recently as just yesterday. This is a generic base module for other modules to use for posting to social media channels.

If we make it more generic like this, and make it available to all content types, then the elections module can build on top of that if there's something it needs that is more specific.

Once we have Social API module installed, we can then install [Social Post](https://www.drupal.org/project/social_post) module (which is what I think elections is using). Social post currently supports posting to

1. [Social Post Facebook](https://www.drupal.org/project/social_post_facebook/) (Update, this module only has dev releases, and seems a bit temperamental. I think we'll probably need to do a bit of work on this to get it working how we want it).
2. [Social Post LinkedIn](https://www.drupal.org/project/social_post_linkedin/) (Update, this module is no longer available)
3. [Social Post Slack](https://www.drupal.org/project/social_post_slack/) (Update, this module is no longer available)
4. [Social Post Twitter](https://www.drupal.org/project/social_post_twitter/)

I guess we could write our own Bluesky integration and contribute it back to the Drupal community.

@willguv do you want to move this issue to the [LocalGov repo](https://github.com/localgovdrupal/localgov) instead?

---
Based on the updates from above, it looks like we can use Social API and Social Post Twitter modules, and then write our own integration for Social Post Bluesky and Social Post LinkedIn.

**Comment by willguv** [comment]
I agree with the approach but would sooner leave the issue here please.

In my experience, social posting has limited use outside Elections and News (maybe there are more microsites uses?) so Elections is the next best opportunity to do the work.

**Comment by willguv** [comment]
We're likely to be using BlueSky in the future (among other services). How long will it take to add this to the Drupal social post module please? It would be nice to give something back.

**Comment by chriswales95** [comment]
@markconroy is correct, we're using the Social API / Social Post module. It only exposes an API.

@willguv Bluesky does have an API of sorts: https://docs.bsky.app/. No PHP native SDK but does have an HTTP API. Not sure how much effort it would be. 

They have a guide on creating a post via HTTP: 
https://docs.bsky.app/blog/create-post
https://docs.bsky.app/docs/api/com-atproto-repo-create-record
